starlette/endpoints.py

```diff
@@ -23,7 +23,12 @@ def __await__(self) -> typing.Generator:
 
     async def dispatch(self) -> None:
         request = Request(self.scope, receive=self.receive)
-        handler_name = "get" if request.method == "HEAD" else request.method.lower()
+        handler_name = (
+            "get"
+            if request.method == "HEAD" and not hasattr(self, "head")
```

Comment:
Is `request.method` always capital `HEAD`?
Or should we consider `if request.method == "HEAD"`?

Reply:
always uppercase https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope